### DISCUSSION TITLE
useSuspenseQuery called twice in Next.js app directory

### DISCUSSION THREAD

**Discussion by Sh031224** [TanStack/query#7325] [Q&A] [answered]
## Problem

useSuspenseQuery is called once on the server and once on the client.

I followed this [example](https://tanstack.com/query/v5/docs/framework/react/guides/advanced-ssr#initial-setup), but I don't know why it gets called twice. 

It seems to only happen when using next-themes, but I haven't been able to figure out the exact cause.
I also looked at the next-themes code, but didn't find any code that could be problematic.

If there is a problem with my configuration itself, I would appreciate it if you could let me know.

## Demo

https://codesandbox.io/p/devbox/next-themes-suspense-error-forked-ls72ls

https://github.com/TanStack/query/assets/51149996/9c032ac0-9404-46f6-a0f0-5a7d48910e09

An issue has also been created in [next-themes](https://github.com/pacocoursey/next-themes/issues/283).

**Comment by TkDodo** [comment, marked as answer]
seems like you are neither using `initialData`, or `hydration`, or using our experimental nextjs streaming adapter. So the data from the server never makes it to the client side cache.

**Reply by Sh031224** [reply]
This is the part I missed. Thank you.
